examples/window/window_settings.rs

```diff
@@ -10,14 +10,14 @@ use bevy::{
 fn main() {
```

Comment:
Dragging windows to resize them is also funky / doesn't behave as expected.

Reply:
I *think* this is caused by setting the resolution after requesting maximization, since the OS is probably thinking by setting the resolution we want to un-maximize.

Reply:
Ya this seems to work for me now, could you confirm @cart ?

Reply:
This is still broken for me. Weirdly resizing from the bottom right hand corner works, but resizing from any other corner causes the window to drift to the upper left hand corner.

Reply:
I'm guessing this is somehow wrong:
https://github.com/bevyengine/bevy/pull/5589/files#diff-67a3a9f85d452e515e78e0407eb4828c19ddac86c50b1646896f8778ade59a12R222-R239

Could you try testing removing the condition on resolution changing and only leave the position change detection and see if it still causes it?

Reply:
I've been struggling with resizing for awhile now. I've already tried that (and it does fix the drift problem). But theres still an issue with rightward drift on resize and weirdly, vertical growing (even when you only scale horizontally). I've been adding as much tracing as I can, but so far nothing.

Reply:
It these issues don't happen on _every_ resize event. Just some of them. Very weird stuff. (and I've confirmed this isn't an issue on `main`)

Reply:
hmm, I'll try to get my hands on my old laptop soon to try and repro it on that.

Reply:
Like where is this vertical resize event coming from? Im only dragging the horizontal/left side of the window. info log is from the `if window.resolution != previous.resolution` part of `changed_window`. Normal println events are from where we initially process the winit window events. Ignore the vulkan error, thats a separate issue / already root caused.  

![image](https://user-images.githubusercontent.com/2694663/213317497-b15c92dc-ad0f-4ea0-872f-8f91a02b3c78.png)

Reply:
(I've added `info` logs in front of every winit window property set op in `changed_window`, so we aren't changing anything else there when this happens)

Reply:
Fixed!  Reading the winit window's size directly when choosing whether or not to set from the Window size is "race-ey", so sometimes theres a feedback loop.
If we instead use the "last seen winit window value according to the event loop", the weirdness stops